src/util/child-component.js

```diff
@@ -18,68 +18,82 @@ export function createChildComponent (definition={}) {
+			 * @return {array} elements with this component type
 			 */
 			getOwnChildren(elements) {
 				let component = this;
-				let children = [];
+				let ownElements = [];
 				React.Children.forEach(elements, (element) => {
 					if (element && element.type === component) {
-						children = children.concat(React.Children.map(element.props.children, x => x));
+						ownElements.push(element);
 					}
 				});
-				return children;
+				return ownElements;
 			},
 
 			/**
-			 * getAll
+			 * getOwnChildrenAsProps
 			 *
-			 * Returns all elements with this component type
+			 * Returns the `props` object of each element with this component type
 			 *
-			 * @param {Array} elements - usually `this.props.children` but can be any list of elements which
-			 *	                          might have this type
-			 * @return {Array} elements with this component type
+			 * @param {array} elements - usually `this.props.children` but can be any list of elements which might have this type
+			 * @return {array} `props` objects of elements with this component type
 			 */
-			getAll(elements) {
+			getOwnChildrenAsProps(elements) {
 				let component = this;
-				let elementsOfThisType = [];
-				React.Children.forEach(elements, (element) => {
-					if (element && element.type === component) {
-						elementsOfThisType = elementsOfThisType.concat(element);
-					}
-				});
-				return elementsOfThisType;
+				return _.map(component.getOwnChildren(elements), 'props');
 			},
 
 			/**
-			 * getAllProps
+			 * getOwnProps
 			 *
-			 * Returns the `props` object of each element with this component type
+			 * Given a set of parent props this will return only the props for this
+			 * component type
 			 *
-			 * @param {Array} elements - usually `this.props.children` but can be any list of elements which
-			 *	                          might have this type
-			 * @return {Array} `props` objects of elements with this component type
+			 * @param {object} parentProps - usually `this.props` that may contain a prop for the current component type
+			 * @return {array} - an array of normalized props
 			 */
-			getAllProps(elements) {
-				let component = this;
-				return _.map(component.getAll(elements), 'props');
-			},
+			getOwnProps(parentProps) {
+				// Extract out the defined `childProps` from the parent class
+				const currentProps = _.chain(parentProps)
+					.get(definition.propName) // grab the prop we care about, e.g. "Child"
+					.thru(x => [x]) // wrap in an array
+					.flatten()
+					.filter(x => x) // remove falsey values
+					.value();
 
-			getAllAsProps(props, elements) {
-				let component = this;
-				return _.map(_.flatten([_.get(props, definition.propName, [])]), (childProp) => {
+				return _.map(currentProps, (childProp) => {
 					if (_.isPlainObject(childProp)) {
-						return _.assign({}, childProp, {children: childProp.children || childProp.text});
+						return childProp;
```

Comment:
I got rid of the `text` alias cause I thought it was confusing. We can always add it back if needed.